File: app/src/main/java/com/lttrung/dormitory/database/data/network/RoomNetwork.java
package com.lttrung.dormitory.database.data.network;

import java.lang.reflect.Type;

import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Single;

@Singleton
public interface RoomNetwork {
    Single<Type> fetchRooms();
    Single<Type> fetchRoomDetails();
}

File: app/src/main/java/com/lttrung/dormitory/database/data/network/RoomNetwork.kt
package com.lttrung.dormitory.database.data.network

import com.lttrung.dormitory.database.data.network.models.Room
import io.reactivex.rxjava3.core.Single
import java.lang.reflect.Type
import javax.inject.Singleton

@Singleton
interface RoomNetwork {
    fun fetchRooms(roomTypeId: Int): Single<List<Room>>
    fun fetchRoomDetails(): Single<Type>
}